/*
 * This file was automatically generated by EvoSuite
 * Sat Apr 22 09:04:19 KST 2017
 */

package com.sleepycat.je.log;

import static org.evosuite.runtime.EvoAssertions.assertThrownBy;
import static org.junit.Assert.assertTrue;
import static org.junit.Assert.fail;

import java.nio.ByteBuffer;
import java.nio.charset.Charset;

import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.junit.Test;
import org.junit.runner.RunWith;

import com.sleepycat.je.dbi.EnvironmentImpl;

import sun.nio.cs.ext.MS949;

@RunWith(EvoRunner.class) @EvoRunnerParameters(useVNET = true) 
public class ChecksumValidator_ESTest3 extends ChecksumValidator_ESTest_scaffolding3 {

  @Test
  public void test00()  throws Throwable  {
      ChecksumValidator checksumValidator0 = new ChecksumValidator();
      try { 
        checksumValidator0.validate((EnvironmentImpl) null, 0, 0, (-1L), false);
        fail("Expecting exception: DbChecksumException");
      
      } catch(DbChecksumException e) {
         //
         // Location 0x0/0xffffffff expected 0 got 1
         //
         assertThrownBy("com.sleepycat.je.log.ChecksumValidator", e);
      }
  }

  @Test
  public void test01()  throws Throwable  {
      ChecksumValidator checksumValidator0 = new ChecksumValidator();
      try { 
        checksumValidator0.validate((EnvironmentImpl) null, (-1323L), (-2788L));
        fail("Expecting exception: DbChecksumException");
      
      } catch(DbChecksumException e) {
         //
         // Location 0xffffffff/0xfffff51c expected -1323 got 1
         //
         assertThrownBy("com.sleepycat.je.log.ChecksumValidator", e);
      }
  }

  @Test
  public void test02()  throws Throwable  {
      ChecksumValidator checksumValidator0 = new ChecksumValidator();
      MS949 mS949_0 = (MS949)Charset.defaultCharset();
      ByteBuffer byteBuffer0 = mS949_0.encode("7.oU 6=h&Z:a?jLcW");
      checksumValidator0.update((EnvironmentImpl) null, byteBuffer0, 0, false);
      assertTrue(byteBuffer0.hasArray());
  }

  @Test
  public void test03()  throws Throwable  {
      ChecksumValidator checksumValidator0 = new ChecksumValidator();
      try { 
        checksumValidator0.validate((EnvironmentImpl) null, 9223372036854775807L, 9223372036854775807L, 9223372036854775807L, true);
        fail("Expecting exception: DbChecksumException");
      
      } catch(DbChecksumException e) {
         //
         // Location 0xffffffff/0xffffffff expected 9223372036854775807 got 1
         //
         assertThrownBy("com.sleepycat.je.log.ChecksumValidator", e);
      }
  }

  @Test
  public void test04()  throws Throwable  {
      ChecksumValidator checksumValidator0 = new ChecksumValidator();
      try { 
        checksumValidator0.validate((EnvironmentImpl) null, 0L, 1L, 1L, false);
        fail("Expecting exception: DbChecksumException");
      
      } catch(DbChecksumException e) {
         //
         // Location 0x1/0x1 expected 0 got 1
         //
         assertThrownBy("com.sleepycat.je.log.ChecksumValidator", e);
      }
  }

  @Test
  public void test05()  throws Throwable  {
      ChecksumValidator checksumValidator0 = new ChecksumValidator();
      checksumValidator0.validate((EnvironmentImpl) null, 1L, 0L, 0L, false);
  }

  @Test
  public void test06()  throws Throwable  {
      ChecksumValidator checksumValidator0 = new ChecksumValidator();
      try { 
        checksumValidator0.validate((EnvironmentImpl) null, 809L, 809L);
        fail("Expecting exception: DbChecksumException");
      
      } catch(DbChecksumException e) {
         //
         // Location 0x0/0x329 expected 809 got 1
         //
         assertThrownBy("com.sleepycat.je.log.ChecksumValidator", e);
      }
  }

  @Test
  public void test07()  throws Throwable  {
      ChecksumValidator checksumValidator0 = new ChecksumValidator();
      checksumValidator0.validate((EnvironmentImpl) null, 1L, 1L);
  }

  @Test
  public void test08()  throws Throwable  {
      ChecksumValidator checksumValidator0 = new ChecksumValidator();
      try { 
        checksumValidator0.update((EnvironmentImpl) null, (ByteBuffer) null, 1279, true);
        fail("Expecting exception: DbChecksumException");
      
      } catch(DbChecksumException e) {
         //
         // null buffer given to checksum validation, probably  result of 0's in log file. true
         //
         assertThrownBy("com.sleepycat.je.log.ChecksumValidator", e);
      }
  }

  @Test
  public void test09()  throws Throwable  {
      ChecksumValidator checksumValidator0 = new ChecksumValidator();
      try { 
        checksumValidator0.update((EnvironmentImpl) null, (ByteBuffer) null, (-1179), false);
        fail("Expecting exception: DbChecksumException");
      
      } catch(DbChecksumException e) {
         //
         // null buffer given to checksum validation, probably  result of 0's in log file. false
         //
         assertThrownBy("com.sleepycat.je.log.ChecksumValidator", e);
      }
  }

  @Test
  public void test10()  throws Throwable  {
      ChecksumValidator checksumValidator0 = new ChecksumValidator();
      MS949 mS949_0 = (MS949)Charset.defaultCharset();
      ByteBuffer byteBuffer0 = mS949_0.encode("");
      // Undeclared exception!
      try { 
        checksumValidator0.update((EnvironmentImpl) null, byteBuffer0, (-1), true);
        fail("Expecting exception: ArrayIndexOutOfBoundsException");
      
      } catch(ArrayIndexOutOfBoundsException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         assertThrownBy("java.util.zip.Adler32", e);
      }
  }

  @Test
  public void test11()  throws Throwable  {
      ChecksumValidator checksumValidator0 = new ChecksumValidator();
      checksumValidator0.reset();
  }
}
